### DISCUSSION TITLE
Not Found | /404.html

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#949] [Comments]
https://yihui.org/404.html

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks for letting me know! There is a missing trailing slash in the link: <https://yihui.name/knitr/options/#cache>

> Originally posted on 2017-07-25 04:32:20

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks a lot! I just fixed it.

> Originally posted on 2019-05-05 02:48:42

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
已经修正。谢谢！

> ——原帖发布于 2019-05-16 03:08:14

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
I just fixed it. Thank you!

> Originally posted on 2019-06-13 03:20:07

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Fixed now. Thanks for the report!

> Originally posted on 2019-09-01 03:18:34

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks for the report! The link works for me. Perhaps it was a temporary issue?

> Originally posted on 2019-10-24 03:04:47

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks! Where did you get this link?

> Originally posted on 2020-01-01 01:09:29

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Could you tell me how to reproduce it? Which link did you click in RStudio? Thanks!

> Originally posted on 2020-01-19 05:58:42

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks, but as I asked below, how did you get that link?

> Originally posted on 2020-01-26 04:29:25

**Reply by yihui** [reply]
I just fixed it. Thank you so much!

> Originally posted on 2020-01-28 16:00:50

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks for letting me know!

> Originally posted on 2020-03-11 20:22:58

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks! This page has not been committed to my Github repo until now.

> Originally posted on 2021-06-16 04:54:45

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks! It should be <https://yufree.cn/cn/2020/02/12/intj/> instead. I just fixed it on the page <https://yihui.org/cn/2020/07/principles-notes/>

> Originally posted on 2020-12-13 03:28:39

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks, but the former link seems to be valid. Perhaps it was a temporary problem.

> Originally posted on 2020-12-31 05:19:05

**Reply by yihui** [reply]
Oh I see! Just fixed it. Thanks!

> Originally posted on 2021-01-01 21:49:48

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
I'll fix it in the next few days. Thanks for the report!

> Originally posted on 2021-06-10 02:21:06

[a comment by an automated account is not shown]

**Comment by chuxinyuan** [comment]
看到 `{{ if and (not .Params.disable_comments) (or (eq .Kind "404") (and (not .IsHome) .Content)) }}` 专门过来看看，我相信我一定不是最后一个专门来这里的。

顺便提一句，在没有 `disable_comments: true` 的时候，404 页面本身就是可以评论的，这里的代码是否可以简化下，烧脑的很啊。

**Reply by yihui** [reply]
`disable_comments: true` 有最高优先级，它可以禁用任何页面的评论区，包括 404 页面。

如果这个参数未设置，那么接下来看：

- 如果是 404 页面，那么启用评论。
- 如果是首页，那么禁用；否则要求页面内容非空，才启用评论，这是为了去除纯列表页面，比如 <https://yihui.org/en/>。

这只是我自己的偏好而已，其它用户不必按照我的偏好设置，也不必读懂我的代码。比如可以任何页面都无条件启用评论区。

**Reply by chuxinyuan** [reply]
我是默认无条件启用评论区，在不需要评论的地方补个刀，比如首页、纯列表页面等。

**Reply by chuxinyuan** [reply]
`{{ if or (and (not .Params.disable_comments) (not .IsHome) .Content) (eq .Kind "404")) }}`

**Reply by yihui** [reply]
这个写法与我原来的写法不等价，404 页面会无条件启用评论，`disable_comments` 参数无法控制它。另外，末尾多了个回括号吧。

**Reply by chuxinyuan** [reply]
我主要是看到你的 404 已经有评论了，你这么爱惜评论，应该不会关闭它吧！恩，总开关得有，还是你那个写法严谨！

**Reply by yihui** [reply]
不会关评论。但写程序的时候通常需要留条后路，尽量不写死。

**Comment by ashishtiwari03** [comment]
hii, https://yihui.org/animation/example/kmeans/  404 not found

**Reply by yihui** [reply]
Thanks! Could you tell me where you got this link? I don't think I have this link on my site. The link to the k-means animation is https://yihui.org/animation/example/kmeans-ani/

**Reply by ashishtiwari03** [reply]
Thank you for pointing that out and for sharing the correct link.

That was my mistake, I accidentally omitted the “-ani” in the URL while following the example link pattern from other entries. I appreciate the correction, and I will update the link to:

https://yihui.org/animation/example/kmeans-ani/

Thanks again for the clarification!
